### PR TITLE
Improving taxonomy page title

### PR DESCRIPTION
**Pull request by syedgalib**

### CODE REVIEW

**Review by vairafiq — APPROVED** [review]
Works fine

File: includes/classes/class-seo.php [hunk 1]
```diff
@@ -105,22 +106,36 @@ public function wpseo_opengraph_title($title)
             return $title;
         }
 
-        public function atbdp_title_update($title, $id = null)
+        public function update_taxonomy_page_title($title, $id = null)
         {
+            if ( is_null( $id ) ) return $title;
```
Comment (obiPlabon) [inline includes/classes/class-seo.php:111]: Please avoid these single-line conditions and make sure to follow WordPress coding standards as much as possible.

```php
if ( is_null( $id ) ) {
    return $title;
}
```

File: includes/classes/class-seo.php [hunk 2]
```diff
@@ -105,22 +106,36 @@ public function wpseo_opengraph_title($title)
...
+
+            return $this->get_taxonomy_page_title( $title, $id );
+        }
+
+        public function update_taxonomy_single_page_title( $title, $post )
+        {
+            if ( is_null( $post ) ) return $title;
```
Comment (obiPlabon) [inline includes/classes/class-seo.php:118]:
```php
if ( is_null( $post ) ) {
    return $title;
}
```

File: includes/classes/class-seo.php [hunk 3]
```diff
@@ -105,22 +106,36 @@ public function wpseo_opengraph_title($title)
...
+
+            return $this->get_taxonomy_page_title( $title, $post->ID );
+        }
+
+        public function get_taxonomy_page_title( $default_title, $page_id )
+        {
+            if ( is_admin() ) { return $default_title; }
```
Comment (obiPlabon) [inline includes/classes/class-seo.php:125]:
```php
if ( is_admin() || ! is_int( $page_id ) ) {
    return $default_title;
}
```

**Review by obiPlabon — APPROVED** [review]
@syedgalib Please check the feedbacks and keep notes for the future.

Thank you

**Pull request merged**